### PR TITLE
More simplification rules for ternary add/sub with division

### PR DESCRIPTION
**Pull request by psuriana**

### CODE REVIEW

**Comment by psuriana** [comment]
@abadams @dsharletg PTAL

File: src/Simplify.cpp [hunk 1]
```diff
@@ -1232,9 +1308,76 @@ class Simplify : public IRMutator {
...
+        } else if (no_overflow(op->type) &&
+                   div_a && sub_div_a_a &&
+                   is_simple_const(div_a->b) &&
+                   equal(b, sub_div_a_a->b)) {
+            // (y - x)/c - x -> (y - (c + 1)*x)/c
+            expr = mutate((sub_div_a_a->a - (div_a->b + 1)*b)/div_a->b);
+        } else if (no_overflow(op->type) &&
                    min_b &&
                    add_b_a &&
-                   no_overflow(op->type) &&
```
Comment (dsharletg) [inline src/Simplify.cpp]: Is this really safe to remove? (And another below.)

File: src/Simplify.cpp [hunk 2]
```diff
@@ -2500,6 +2641,54 @@ class Simplify : public IRMutator {
                    equal(add_a->b, add_b->a)) {
             // min(a + b, b + c) -> min(a, c) + b
             expr = mutate(min(add_a->a, add_b->b)) + add_a->b;
+        } else if (no_overflow(op->type) &&
```
Comment (dsharletg) [inline src/Simplify.cpp]: Are these all permutations of commuting/reassociating the same rule? I'm fine with it... just curious... we should try to think about a better way to do this (in a separate PR). I know we have other instances of rules like this too.

File: src/Simplify.cpp (hunk 1 above)
Comment (psuriana) [inline src/Simplify.cpp]: It already check for no_overflow at line 1378

File: src/Simplify.cpp (hunk 2 above)
Comment (psuriana) [inline src/Simplify.cpp]: Yeah, these are some of the permutations (not all of them).

File: src/Simplify.cpp (hunk 1 above)
Comment (dsharletg) [inline src/Simplify.cpp]: Ah of course, funny bug.

**Pull request merged**